### DISCUSSION TITLE
Support curveStep in link generators

### DISCUSSION THREAD

**Discussion by RyKilleen** [d3/d3#3481] [Q&A] [answered]
It's been a few years since I've revisited D3, and I'm excited to catch up on all the new features!

One thing I was surprised to find with link generators is that the Bezier curves are pretty rigid (maybe that's necessary, I don't know!) Would there be any interest in supporting `curveStep` etc in the different link generators, if that's a feasible thing? 

ie something like `linkHorizontal().x(...).y(...).curve(d3.curveStep)` ?

I might be missing something very fundamental, spending the weekend shaking the rust off currently.

**Comment by mbostock** [comment]
Have you seen the recently-added bump curves?

https://github.com/d3/d3-shape#curveBumpX

Edit: oops, I see you were asking for the inverse.

**Comment by Fil** [comment, marked as answer]
See https://github.com/d3/d3-shape/pull/177
